src/sentence-length.ts

```diff
@@ -45,16 +51,32 @@ const defaultOptions: Required<Options> = {
     /**
      * @deprecated
      */
-    exclusionPatterns: []
+    exclusionPatterns: [],
+    countBy: "code units"
 };
 
 const isSentenceNode = (node: TxtParentNodeWithSentenceNodeContent): node is TxtSentenceNode => {
     return node.type === SentenceSplitterSyntax.Sentence;
 };
+/**
+ * A count of the number of codepoint currently in the string.
+ *
+ * Complexity: O(n)
+ * @param s string
+ */
+const strLenByCodePoint = (s: string): number => {
+    let i = 0;
+    for (const _ of s) {
+        ++i;
+    }
+    return i;
+};
 const reporter: TextlintRuleReporter<Options> = (context, options = {}) => {
     const maxLength = options.max ?? defaultOptions.max;
     const skipPatterns = options.skipPatterns ?? options.exclusionPatterns ?? defaultOptions.skipPatterns;
     const skipUrlStringLink = options.skipUrlStringLink ?? defaultOptions.skipUrlStringLink;
+    const strLen =
+        options.countBy == null || options.countBy === "code units" ? (s: string) => s.length : strLenByCodePoint;
```

Comment:
Can you create a function like `strLenByCodeUnits` and use it?

```js
const countBy = options?.countBy ?? defaultOptions.countBy;
const strLen = countBy === "codeunits" ? strLenByCodeUnits : strLenByCodePoint;
```

Reply:
c555591520112fb843b44c387fbf3578f8ed43c8